src/Stratis.Features.FederatedPeg/TargetChain/SignedMultisigTransactionBroadcaster.cs

```diff
@@ -42,104 +35,70 @@ public class SignedMultisigTransactionBroadcaster : ISignedMultisigTransactionBr
-        {
-            this.asyncLoop = this.asyncProvider.CreateAndRunAsyncLoop(nameof(PartialTransactionRequester), _ =>
-                {
-                    this.BroadcastTransactionsAsync().GetAwaiter().GetResult();
-                    return Task.CompletedTask;
-                },
-                this.nodeLifetime.ApplicationStopping,
-                TimeBetweenQueries);
-        }
-
-        /// <inheritdoc />
-        public async Task BroadcastTransactionsAsync()
+        private async Task OnCrossChainTransactionFullySigned(CrossChainTransferTransactionFullySigned @event)
         {
             if (this.ibdState.IsInitialBlockDownload() || !this.federationWalletManager.IsFederationWalletActive())
             {
                 this.logger.LogTrace("Federation wallet isn't active or in IBD. Not attempting to broadcast signed transactions.");
                 return;
             }
 
-            ICrossChainTransfer[] transfers = this.store.GetTransfersByStatus(new CrossChainTransferStatus[]{CrossChainTransferStatus.FullySigned});
-
-            if (!transfers.Any())
+            TxMempoolInfo txInfo = await this.mempoolManager.InfoAsync(@event.Transfer.PartialTransaction.GetHash()).ConfigureAwait(false);
+            if (txInfo != null)
             {
-                this.logger.LogTrace("Signed multisig transactions do not exist in the CrossChainTransfer store.");
+                this.logger.LogTrace("Deposit ID '{0}' already in the mempool.", @event.Transfer.DepositTransactionId);
                 return;
             }
 
-            foreach (ICrossChainTransfer transfer in transfers)
-            {
-                TxMempoolInfo txInfo = await this.mempoolManager.InfoAsync(transfer.PartialTransaction.GetHash()).ConfigureAwait(false);
-
-                if (txInfo != null)
-                {
-                    this.logger.LogTrace("Deposit ID '{0}' already in the mempool.", transfer.DepositTransactionId);
-                    continue;
-                }
+            this.logger.LogInformation("Broadcasting deposit-id={0} a signed multisig transaction {1} to the network.", @event.Transfer.DepositTransactionId, @event.Transfer.PartialTransaction.GetHash());
 
-                this.logger.LogInformation("Broadcasting deposit-id={0} a signed multisig transaction {1} to the network.", transfer.DepositTransactionId, transfer.PartialTransaction.GetHash());
+            await this.broadcasterManager.BroadcastTransactionAsync(@event.Transfer.PartialTransaction).ConfigureAwait(false);
+        }
 
-                await this.broadcasterManager.BroadcastTransactionAsync(transfer.PartialTransaction).ConfigureAwait(false);
-            }
+        /// <inheritdoc />
+        public void Start()
+        {
+            this.onCrossChainTransactionFullySignedSubscription = this.signals.Subscribe<CrossChainTransferTransactionFullySigned>(async (tx) => await this.OnCrossChainTransactionFullySigned(tx).ConfigureAwait(false));
```

Comment:
why not pass this to the delegate `OnCrossChainTransactionFullySigned`?

Reply:
you mean why not having passed this.OnCrossChainTransactionFullySigned directly? To have it async, because we don't care about completing this before returning to whoever called publish